plugin/src/helpers/config.ts

```diff
@@ -65,14 +65,25 @@ function hasPlugin(plugins: PluginRef[], pluginName: string): boolean {
   )
 }
 
-export function checkGatsbyConfig({ utils, netlifyConfig }): void {
+export async function checkConfig({ utils, netlifyConfig }): Promise<void> {
+  const gatsbyRoot = getGatsbyRoot(netlifyConfig.build.publish)
+
   // warn if gatsby-plugin-netlify is missing
   const gatsbyConfig = loadGatsbyConfig({
     utils,
-    publish: netlifyConfig.build.publish,
+    gatsbyRoot,
   })
 
-  if (!hasPlugin(gatsbyConfig.plugins, 'gatsby-plugin-netlify')) {
+  if (hasPlugin(gatsbyConfig.plugins, 'gatsby-plugin-netlify')) {
+    if (
+      // prettier-ignore
```

Comment:
FYI this ignore is just to avoid tripping over the max-lines lint rule!

Reply:
Why does it need a prettier ignore for that? Isn't that an eslint rule?

Reply:
the file was under 150 lines until prettier formatted that 1 line onto 3. I tried ignoring prettier to see if it passed the eslint rules and it did, so I just kept it there. I can swap it for the eslint-ignore if you think it makes more sense?

Reply:
Yeah, just do the eslint ignore for max lines for the whole file